libsoftwarecontainer/unit-test/networkgateway_unittest.cpp

```diff
@@ -60,14 +122,268 @@ TEST_F(NetworkGatewayTest, TestNoContainer) {
  */
 TEST_F(NetworkGatewayTest, TestSetConfig) {
     givenContainerIsSet(gw);
-    const std::string config = "[{\"internet-access\": false, \"gateway\":\"\"}]";
+
+    ASSERT_TRUE(gw->setConfig(VALID_FULL_CONFIG));
+}
+
+/**
+ *@brief Test that config entries with no priority specified fails gracefully.
+ */
+TEST_F(NetworkGatewayTest, TestSetConfigNoPriority) {
+    givenContainerIsSet(gw);
+
+    const std::string config =
+    "[{"
+        "\"type\": \"OUTGOING\","
+        "\"rules\": ["
+                     "{ \"host\": \"127.0.0.1/16\", \"port\": 80, \"target\": \"ACCEPT\"},"
+                     "{ \"host\": \"google.com\", \"port\": \"80-85\", \"target\": \"ACCEPT\"},"
+                     "{ \"host\": \"127.0.0.1/16\", \"port\": [80, 8080], \"target\": \"ACCEPT\"},"
+                     "{ \"host\": \"50.63.202.33/24\", \"target\": \"REJECT\"}"
+                 "],"
+        "\"default\": \"ACCEPT\""
+    "}]";
+
+    ASSERT_FALSE(gw->setConfig(config));
+}
+
+/**
+ * @brief Test that config entries with no type specified fails gracefully.
+ */
+TEST_F(NetworkGatewayTest, TestSetConfigNoType) {
+    givenContainerIsSet(gw);
```

Comment:
So when I read that method call, I read "given that a container is set". I don't know what that means. The call just adds a gateway to a container right? So, then perhaps givenAddingThisGateway() or something would make sense. Or how should I read the method as it is now?